course-material/intro-to-sql/combining-data-from-multiple-tables.md

```diff
@@ -1,14 +1,305 @@
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.card AS
+FROM read_csv_auto('expanded_data/card.csv', header=True, sep=',');
+CREATE TABLE s1.link AS
+FROM read_csv_auto('expanded_data/disp.csv', header=True, sep=',');
+```
+
+## Renaming Columns
+
+Before we start joining the above ingested tables, we need to rename the columns of the `district` table for better readability. We can do this by using the `ALTER TABLE` syntax in DuckDB. The `RENAME COLUMN` syntax allows us to rename a column in a table.
+
+```{tip}
+We can also rename multiple columns in a table by separating the column names with a comma, which is more efficient. However, this method is not supported by DuckDB. Be sure to look at the documentation of the database you are using to see if it supports this feature.
+```
+
+```{code-cell} ipython3
+%%sql
+ALTER TABLE s1.district
+RENAME COLUMN A1 TO district_id;
+ALTER TABLE s1.district
+RENAME COLUMN A2 TO district_name;
+ALTER TABLE s1.district
+RENAME COLUMN A3 TO region;
+ALTER TABLE s1.district
+RENAME COLUMN A4 TO no_of_inhabitants;
+ALTER TABLE s1.district
+RENAME COLUMN A5 TO no_of_municipalities_lt_499;
+ALTER TABLE s1.district
+RENAME COLUMN A6 TO no_of_municipalities_500_1999;
+ALTER TABLE s1.district
+RENAME COLUMN A7 TO no_of_municipalities_2000_9999;
+ALTER TABLE s1.district
+RENAME COLUMN A8 TO no_of_municipalities_gt_10000;
+ALTER TABLE s1.district
+RENAME COLUMN A9 TO no_of_cities;
+ALTER TABLE s1.district
+RENAME COLUMN A10 TO ratio_of_urban_inhabitants;
+ALTER TABLE s1.district
+RENAME COLUMN A11 TO average_salary;
+ALTER TABLE s1.district
+RENAME COLUMN A12 TO unemployment_rate_95;
+ALTER TABLE s1.district
+RENAME COLUMN A13 TO unemployment_rate_96;
+ALTER TABLE s1.district
+RENAME COLUMN A14 TO no_of_entrepreneurs_per_1000_inhabitants;
+ALTER TABLE s1.district
+RENAME COLUMN A15 TO no_of_committed_crimes_95;
+ALTER TABLE s1.district
+RENAME COLUMN A16 TO no_of_committed_crimes_96;
+```
+
+## Combining All Tables
+
+<!-- Now that we have renamed the columns of the `district` table, we can now combine the `account`, `card`, `district`, and `link` tables. We can do this by using the `CREATE VIEW` syntax in DuckDB. The `CREATE VIEW` syntax allows us to create a virtual table that is a result of a query. 
+
+```{important}
+The `CREATE VIEW` syntax is not supported by all databases. Be sure to look at the documentation of the database you are using to see if it supports this feature.
+```
+
+```{code-cell} ipython3
+%%sql   
+CREATE VIEW s1.combined_tables AS
```

Comment:
The inner join was covered in @jpjon's tutorial. 

Can you suggest other ways to combine data?

Maybe we can focus the inner, outer and other joins in your tutorial to allow for a more comprehensive set of examples?

Reply:
ok sure, can focus more on natural and outer joins